token/program/src/processor.rs

```diff
@@ -644,6 +644,36 @@ impl Processor {
         Ok(())
     }
 
+    /// Processes a [SyncNative](enum.TokenInstruction.html) instruction
+    pub fn process_sync_native(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        let native_account_info = next_account_info(account_info_iter)?;
+
+        if native_account_info.owner != program_id {
+            return Err(ProgramError::IncorrectProgramId);
+        }
+        let mut native_account = Account::unpack(&native_account_info.data.borrow())?;
+        if native_account.mint != crate::native_mint::id() {
+            return Err(TokenError::NonNativeNotSupported.into());
+        }
+
+        if let COption::Some(rent_exempt_reserve) = native_account.is_native {
```

Comment:
These two checks can be combined, right?

Reply:
Yeah most likely. This was an "abundance of caution" route as with the last PR to be totally sure.  So I'm sure that I understand, do you want me to remove the first check and assume that if this succeeds, we're good?

Reply:
Fwiw, I think just using the second check is consistent with the other processor methods, which use the `is_native()` helper in isolation.

Reply:
Some comments around being careful when assigning `is_native` are probably warranted, but as Tyera mentioned trusting it is consistent.  I checked all the assignments and they appear to do the RightThing :tm: today.

I also like the new error, seems fine to use it in the `None` case instead of the less descriptive `InvalidState`

Reply:
Sounds good! Done in 32d2fbb